src/stackdriver-nozzle/firehose/client.go

```diff
@@ -25,30 +30,38 @@ func NewClient(apiAddress, username, password string, skipSSLValidation bool) Cl
 			SkipSslValidation: skipSSLValidation}}
 }
 
-func (fc *client) StartListening(nozzle logging.Logging) error {
-	cfClient := cfclient.NewClient(fc.cfConfig)
+func (c *client) StartListening(fh FirehoseHandler) error {
+	cfConfig := &cfclient.Config{
+		ApiAddress:        c.cfConfig.ApiAddress,
+		Username:          c.cfConfig.Username,
+		Password:          c.cfConfig.Password,
+		SkipSslValidation: c.cfConfig.SkipSslValidation}
+	cfClient := cfclient.NewClient(cfConfig)
 
-	cachingClient := caching.NewCachingEmpty()
+	cfConsumer := consumer.New(
+		cfClient.Endpoint.DopplerEndpoint,
+		&tls.Config{InsecureSkipVerify: c.cfConfig.SkipSslValidation},
+		nil)
 
-	eventRouter := eventRouting.NewEventRouting(cachingClient, nozzle)
-
-	//err := eventRouter.SetupEventRouting("LogMessage,ValueMetric,HttpStartStop,CounterEvent,Error,ContainerMetric")
-	err := eventRouter.SetupEventRouting("HttpStartStop,ValueMetric")
-	if err != nil {
-		return err
-	}
+	refresher := CfClientTokenRefresh{cfClient: cfClient}
+	cfConsumer.SetIdleTimeout(time.Duration(30) * time.Second)
+	cfConsumer.RefreshTokenFrom(&refresher)
+	messages, errs := cfConsumer.Firehose("test", "")
```

Comment:
We don't need to specify an auth token since we are specifying a token refresher. If the auth token is empty, it will retrieve it from the auth token refresher. Yes, we could make the subscription ID configurable. The subscription ID could be added to the configuration.